src/lib/db.ts

```diff
@@ -0,0 +1,65 @@
+import { API_URL } from './dbconfig';
+import { AuthTokens, fetchAuthSession } from 'aws-amplify/auth';
+
+interface FetchBackendOptions {
+  endpoint: string;
+  method: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH';
+  data?: Record<string, unknown>;
+  authenticatedCall?: boolean;
+}
+
+async function currentSession(): Promise<AuthTokens | null> {
+  try {
+    const res = (await fetchAuthSession()).tokens;
```

Comment:
Possibly needs to use runWithAmplifyServerContext (see middleware.ts for an example), Amplify requires it with Next.js because it runs on server-side by default. (If it doesn't work you may also need to import from aws-amplify/auth/server library.)

Reply:
Should work – tested locally already and was able to fetch authenticated user, session keys, and event from local backend.